### DISCUSSION TITLE
Add way to case-sensitive string filtering

### DISCUSSION THREAD

**Discussion by LukasKalbertodt** [meilisearch/product#418] [Feedback & Feature Proposal]
Equality filters on strings work case-insensitively. This is useful for many cases, but case-sensitive matching is important as well. [The docs](https://docs.meilisearch.com/learn/advanced/filtering_and_faceted_search.html) even mention a use case: 

> Filters have several use-cases, such as restricting the results a specific user has access to [...]

That's also the case for us. Specifically, each item in the index has a list of roles attached to it (a string array). Each user also has a list of roles. A specific user can view a specific item iff at least one of the user's role is inside the item's roles. This can almost be implemented with Meili, if it weren't for the case insensitive check. Roles must certainly be checked case sensitively.

Possible workarounds: doing the filtering in our backend after receiving all results OR encoding the roles somehow, e.g. as hex string. Both options are obviously not great.

So yeah, it would be really great if Meili could filter case sensitively!

**Comment by gmourier** [comment]
Hi @LukasKalbertodt 👋

Why case sensitivity is critical here?

Thanks

**Reply by LukasKalbertodt** [reply]
Well, the reason is simply that our role system works like that: roles are opaque, case-sensitive strings. `foo` and `Foo` are semantically different roles and a user with the role `Foo` is not allowed to see an item with role `foo`.

I think the "don't interpret those strings, just treat them as opaque data" is not too uncommon of a requirement. You could also imagine some base64 encoded binary data in strings. And for base64, lowercase and uppercase letters are obviously different and case-insensitive comparison don't make sense here.

**Reply by gmourier** [reply]
Thanks, @LukasKalbertodt for this precision.

One idea: It could be a new setting permitting to set some attributes to be case-sensitive since case-insensitivity seems to work well for most of our users.

I leave this discussion open to gather more feedback on this and see later if it is to be prioritized.

**Comment by gmourier** [comment]
Hey @LukasKalbertodt 👋

I'm locking that discussion in favor of https://github.com/meilisearch/product/discussions/555

Thank you!
